### ISSUE TITLE
Implement sort feature in All Delegates page

### ISSUE THREAD

**Issue by danieljames-dj** [thewca/worldcubeassociation.org#9081]
This is a React-only task. The task is to bring sort feature in [DelegatesTable.jsx](https://github.com/thewca/worldcubeassociation.org/blob/main/app/webpacker/components/Delegates/DelegatesTable.jsx).

Also, the sorting has to be enabled only when `isAllNonLeadDelegates` is `true` and hence the sorting will be available only in [All Delegates page](https://www.worldcubeassociation.org/delegates#all) under 'Other Delegates' which is restricted only to few users.

(Later, it can be expanded to other pages if requirement comes)

Feel free to take it up if anybody is willing to work on this. (Please leave a comment here before starting so that we won't have multiple people work on same task)

**Comment by dunkOnIT** [comment]
@danieljames-dj is `good first issue` an appropriate label given the complexity of this task?

**Comment by danieljames-dj** [comment]
Yes it is a good first issue. It can be done by following [this page](https://react.semantic-ui.com/collections/table/#variations-sortable) of react-semantic-ui. Thanks for adding the tag.

**Comment by dunkOnIT** [comment]
@sushantshah-dev this might be a good candidate if you're looking to work on something - not particularly complex, but that's the intention with a first issue.

**Comment by bohdancho** [comment]
working on it right now

**Comment by bohdancho** [comment]
Sorry folks, unfortunately I won't be able to finish this PR. Anyone is free to start working on the issue.
